packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -1098,21 +1116,54 @@ export class DatabaseCluster extends DatabaseClusterNew {
   constructor(scope: Construct, id: string, props: DatabaseClusterProps) {
     super(scope, id, props);
 
-    const credentials = renderCredentials(this, props.engine, props.credentials);
-    const secret = credentials.secret;
+    if (
+      props.manageMasterUserPassword &&
+      (
+        props.credentials?.excludeCharacters ||
```

Comment:
It might already be taken care of further up, but `ManageMasterUserPassword` and `MasterUserPassword` are mutually exclusive, so would be good to throw if they accidentally set both. 

When I was manually setting `ManageMasterUserPassword` I had to intentionally remove the `MasterUserPassword` attribute. 

```
        var cfnCluster = (CfnDBCluster)dbCluster.Node.DefaultChild;
        cfnCluster.AddPropertyOverride("ManageMasterUserPassword", true);
        cfnCluster.AddPropertyDeletionOverride("MasterUserPassword");
```

Reply:
Sorry almost forgot, because one of my clusters already existed and I was migrating it from manual password to RDS managed, I also had to blank out the username.

`cfnCluster.AddPropertyDeletionOverride("MasterUsername");`

Reply:
From the testing I did previously, the MasterUsername is respected by the cluster properly when using ManageMasterUserPassword. So both can be used in conjunction.